src/responses/Albany_SolutionTwoNormResponseFunction.cpp

```diff
@@ -146,3 +146,71 @@ evaluateDistParamDeriv(
     dg_dp->assign(0.0);
   }
 }
+
+void
+Albany::SolutionTwoNormResponseFunction::
+evaluateDistParamHessVecProd_xx(
+    const double current_time,
+    const Teuchos::RCP<const Thyra_MultiVector>& v,
+    const Teuchos::RCP<const Thyra_Vector>& x,
+    const Teuchos::RCP<const Thyra_Vector>& xdot,
+    const Teuchos::RCP<const Thyra_Vector>& xdotdot,
+    const Teuchos::Array<ParamVec>& param_array,
+    const Teuchos::RCP<Thyra_MultiVector>& Hv_dp)
+{
+  if (!Hv_dp.is_null()) {
+    Hv_dp->assign(0.0);
```

Comment:
This is wrong, if I understand correctly what's happening. Hxx = d^2 norm(x)/ dx^2, which is not 0. The 1st derivative is x / |x|, the second derivative should be `1/||x|| *Id + x*x^T / ||x||^3 = Id/g + xx^T/g^3`, where Id is the identity and g is the response. So when you apply it to a vector v, the result is `Hxx_v = v/g + (v,x) x/g^3`, where (.,.) is the usual dot product. One dot product plus some saxby should do the trick.

Reply:
I see what you mean. Thanks for highlighting this issue!